test/src/components/rule.js

```diff
@@ -0,0 +1,90 @@
+const assert = require('assert');
+const Rule = require('../../../src/components/rule');
+const SG = require('../../../src/main');
+const world = new SG.World();
+const Thing = SG.Thing;
+const Type = SG.Type;
+const c = SG.constants;
+
+describe('Rule', () => {
+  const world = new SG.World();
+  const animal = new Type('animal');
+  const cat = animal.extend('cat');
+  const dog = animal.extend('dog');
+  const cute = type => type.extend('cute');
+  const fluffy = type => type.extend('fluffy');
+  world.addLocation({ name: 'the garden' });
+  const Sport = world.addThing(new Thing({
+    type: cute(cat),
+    name: 'Sport',
+    locations: ['the garden']
+  }));
+  const Flex = world.addThing(new Thing({
+    type: fluffy(dog),
+    name: 'Flex',
+    locations: ['the garden']
+  }));
+  var rule;
```

Comment:
That's a good question.... maybe? I'll look into that. My immediate thought is that in fact Rule is not a simple class because it is responsible for modelling potential interactions and their results. I imagine a lot of value could be gained from having a lot of tests here. On the other hand, it could be the case that most of the heavy lifting with respect to Rules is actually done by the World module and it's various sub-modules. I guess we'll find out!